### PR TITLE
INFRA-8460 Initial kabuild container

### PR DESCRIPTION
**Pull request by yogieric**
* All dependencies necessary to run "make deps" in webapp
  * "make run" maps ssh-key & ~/khan appropriately to access git, get webapp, etc.
* Try to do everything that linux-setup.sh does
* Enough debug tools that this container is useful for debugging things
    
Issue: https://khanacademy.atlassian.net/browse/INFRA-8460
    
Test plan:
    
Run the following commands:
- make build
- make run_dirty
    
Check that ~/khan contains local ~/khan and that git works.
This is likely very slow on a mac, so be careful.
    
Ideally, try a clean checkout of webapp somewhere followed by
"make deps". This is probably best tested on a linux machine.
    
Also test "make run"

### CODE REVIEW

**Review by dbraley — APPROVED** [review]
Ok, took me a bit to wrap my head around the idea that this base image is FOR webapp, but not in any way bound to it. If it were, this wouldn't feel correct, but as it stands it can be used by other things as well, and is just general infrastructure that could be used for anything, but is presently only used for webapp. I can see that value to that.

Seems maybe a bit odd to copy the aliases into the container, but 🤷🏻 I can't think of a good reason not to do that, I just don't usually see it done.

tldr - lgtm

**Comment by yogieric** [comment]
> Seems maybe a bit odd to copy the aliases into the container, but 🤷🏻 I can't think of a good reason not to do that, I just don't usually see it done.

True. This started as a "debug" container -- and I wanted certain aliases. I'll go back and review that bit.
